lib/utils.js

```diff
@@ -3,196 +3,299 @@ var assert = require('assert');
 var moment = require('moment');
 var vumigo = require('vumigo_v02');
 var Choice = vumigo.states.Choice;
-var JsonApi = vumigo.http.api.JsonApi;
+var _ = require('lodash');
 
 // JS-BOX SPECIFIC UTILS
-var Utils = {
+
 
 // FIXTURES HELPERS
 
-    check_fixtures_used: function(api, expected_used) {
-        var fixts = api.http.fixtures.fixtures;
-        var fixts_used = [];
-        fixts.forEach(function(f, i) {
-            f.uses > 0 ? fixts_used.push(i) : null;
-        });
-        assert.deepEqual(fixts_used, expected_used);
-    },
+check_fixtures_used = function(api, expected_used) {
+    var fixts = api.http.fixtures.fixtures;
+    var fixts_used = [];
+    fixts.forEach(function(f, i) {
+        f.uses > 0 ? fixts_used.push(i) : null;
+    });
+    assert.deepEqual(fixts_used, expected_used);
+};
 
 // TIMEOUT HELPERS
 
-    timed_out: function(im) {
-        return im.msg.session_event === 'new'
-            && im.user.state.name
-            && im.config.no_timeout_redirects.indexOf(im.user.state.name) === -1;
-    },
+timed_out = function(im) {
+    return im.msg.session_event === 'new'
+        && im.user.state.name
+        && im.config.no_timeout_redirects.indexOf(im.user.state.name) === -1;
+};
+
+timeout_redirect = function(im) {
+    return im.config.timeout_redirects.indexOf(im.user.state.name) !== -1;
+};
+
+// NUMBER HELPERS
 
-    timeout_redirect: function(im) {
-        return im.config.timeout_redirects.indexOf(im.user.state.name) !== -1;
-    },
+// An attempt to solve the insanity of JavaScript numbers
+check_valid_number = function(content) {
+    var numbers_only = new RegExp('^\\d+$');
+    return content !== ''
+        && numbers_only.test(content)
+        && !Number.isNaN(Number(content));
+};
 
+double_digit_number = function(input) {
+    input_numeric = parseInt(input, 10);
+    if (parseInt(input, 10) < 10) {
+        return "0" + input_numeric.toString();
+    } else {
+        return input_numeric.toString();
+    }
+};
 
-// SERVICE API CALL HELPERS
+check_number_in_range = function(input, start, end) {
+    return this.check_valid_number(input) &&
+        (parseInt(input, 10) >= start) &&
+        (parseInt(input, 10) <= end);
+};
 
-    service_api_call: function (service, method, params, payload, endpoint, im) {
+// MSISDN HELPERS
 
-        http = new JsonApi(im, {
-            headers: {
-                'Authorization': ['Token ' + im.config.services[service].api_token]
-            }
-        });
+// Check that it's a number and starts with 0 and approximate length
+// TODO: refactor to take length, explicitly deal with '+'
```

Comment:
Please pass in minimum and max length options here, we'll do the complicated things later.  But open an issue for that and include the issue number here.  Like `TODO #6: handle '+'`